.github/workflows/main.yml

```diff
@@ -34,8 +34,8 @@ jobs:
 
   deploy:
     name: Upload binary
-    environment: development
-    concurrency: deploy_dev
+    environment: production
```

Comment:
Seems like cant rename environment https://github.com/orgs/community/discussions/64081

So changing this to production and then main branch can be added to production environment  protection rules